apps/happy-blocks/src/pricing-plans/components/block-settings.tsx

```diff
@@ -0,0 +1,36 @@
+import { InspectorControls } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { SelectControl, PanelRow, PanelBody } from '@wordpress/components';
+import { FunctionComponent } from 'react';
+import { BlockAttributes, PricingPlansConfiguration } from '../types';
+
+interface Props {
+	plans: PricingPlansConfiguration;
+}
+
+const BlockSettings: FunctionComponent<
+	Pick< BlockEditProps< BlockAttributes >, 'attributes' | 'setAttributes' > & Props
+> = ( { attributes, setAttributes, plans } ) => {
+	const options = Object.values( plans ).map( ( plan ) => ( {
+		value: plan.billing.slice( -1 )[ 0 ].planSlug,
```

Comment:
This looks a little bit dangerous. Do you think it's worth adding some checks here?